### PR TITLE
Add option to apply constrained reading in ReadGridded

### PR DESCRIPTION
**Pull request by jgliss**
One usage example is:

Read Angstrom Exponent from a satellite and invalidate all grid points where the AOD is smaller than a certain threshold (e.g. 0.1).

Should be implemented in `ReadGridded.read_var`, e.g. via:

```python
import pyaerocom as pya
reader = pya.io.ReadGridded('AATSR_SU_v4.3')
read_constraint = dict(
   var_name = 'od550aer',
   operator='>',
   filter_val=0.1
)
ae = reader.read_var('ang4487aer', constraints=[read_constraint])
```

### CODE REVIEW

**Comment by jgliss** [comment]
@ejgal this might be interesting for you too. I need this to work quickly, so I will net spend too much efforts for now, but make it work first.

**Comment by jgliss** [comment]
@ejgal would be awesome if you could very briefly have a look at this. As I mentioned, this was done very much in a rush today, but I think it should work,,, and even managed some tests.

File: pyaerocom/io/readgridded.py [hunk 1]
```diff
@@ -1389,11 +1397,132 @@ def _eval_vert_which_and_ts_type(self, var_name, vert_which, ts_type):
...
+            DESCRIPTION.
+
+        Raises
+        ------
+        ValueError
+            DESCRIPTION.
+
+        Returns
+        -------
+        None.
```
Comment (ejgal) [inline pyaerocom/io/readgridded.py:1448]: Could be improved. Return type is wrong.

File: pyaerocom/io/readgridded.py [hunk 2]
```diff
@@ -1449,6 +1578,8 @@ def read_var(self, var_name, start=None, stop=None,
             only relevant if `var_name` is not available for reading but needs
             to be computed: custom method for computation (cf.
             :func:`add_aux_compute` for details)
+        constraints : list, optional
+            list of reading cons
```
Comment (ejgal) [inline pyaerocom/io/readgridded.py:1582]:
```suggestion
            list of reading constraints
```

File: pyaerocom/io/readgridded.py [hunk 3]
```diff
@@ -1389,11 +1397,132 @@ def _eval_vert_which_and_ts_type(self, var_name, vert_which, ts_type):
...
+        mask = operator_fun(other_data.cube.data,
+                            constraint['filter_val'])
+
+        data.cube.data[mask] = new_val
+        return data
+
+    def _try_read_var(self, var_name, start, stop,
+                  ts_type, experiment, vert_which,
+                  flex_ts_type, prefer_longer, **kwargs):
+        """Helper method used in :func:`read_var`
```
Comment (ejgal) [inline pyaerocom/io/readgridded.py:1480]: `_try_read_var` is  just some refactoring, right?

File: pyaerocom/io/readgridded.py [hunk 4]
```diff
@@ -1389,11 +1397,132 @@ def _eval_vert_which_and_ts_type(self, var_name, vert_which, ts_type):
...
+        Returns
+        -------
+        None.
+
+        """
+        self.check_constraint_valid(constraint)
+        if 'new_val' in constraint:
+            new_val = constraint['new_val']
+        else:
+            new_val = np.ma.masked
```
Comment (ejgal) [inline pyaerocom/io/readgridded.py:1455]: Is `new_val` always meant to be np.masked, np.nan or similar, or can it also be some finite value?

Comment (jgliss) [reply pyaerocom/io/readgridded.py:1455]: The idea was that it can be any value

File: pyaerocom/io/readgridded.py (hunk 3 above)
Comment (jgliss) [reply pyaerocom/io/readgridded.py:1480]: Correct

**Comment by jgliss** [comment]
@ejgal hope this looks okay now.

**Comment by ejgal** [comment]
Yes, great!

**Pull request merged**